ui/src/pages/ShoeMap.tsx

```diff
@@ -89,8 +89,34 @@ const StyledLogo = styled.div`
 export const ShoeMap: React.FC = () => {
   const zoom = 4;
   const minZoom = 4;
-  const maxZoom = 18;
+  const maxZoom = 12;
+  const timeoutSeconds = 1728000000;
   const currentLocation = { lat: 53.655697, lng: -100.13316 };
+  const key = "timestamp";
+
+  const setTimeStamp = () => {
+    const item = {
+      value: "value",
+      expiry: new Date().getTime(),
+    };
+    localStorage.setItem(key, JSON.stringify(item));
+  };
+
+  const compareTimestamp = () => {
+    const oldTimestampStr = localStorage.getItem(key);
+    if (!oldTimestampStr) {
+      setTimeStamp();
+      return TutorialState.First;
+    }
+    const oldTimestamp = JSON.parse(oldTimestampStr);
+    const curTimestamp = new Date();
+    if (curTimestamp.getTime() - oldTimestamp.expiry > timeoutSeconds) {
+      localStorage.removeItem(key);
+      setTimeStamp();
+      return TutorialState.First;
+    }
+    return TutorialState.Closed;
```

Comment:
If you can, please write a test to confirm that these functions work as expected.

Reply:
![jesttests](https://user-images.githubusercontent.com/19617248/101848669-0ee9d900-3b24-11eb-9327-acda69a32b79.png)